### ISSUE TITLE
Multipass instance won't start after upgrade

### ISSUE THREAD

**Issue by csandwith** [canonical/multipass#2371]
(Possibly related to https://github.com/canonical/multipass/issues/2316)

This morning, Ubuntu let me know there were software updates available - I did them and had to restart the machine to complete them - now that I have, my multipass instances will not start, having tried multiple times.

Sometimes I get "timed out waiting for response", sometimes I get "start failed: failed to start qemu instance: program: qemu-system-x86_64; error: Process operation timed out"

```
sudo lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal

multipass --version
multipass   1.8.0
multipassd  1.8.0
```

Here's the journalctl output after a failed start ("timed out waiting for response"):
```
Dec 08 09:09:56 cody-MS-7850 multipassd[26866]: [27376] started: /snap/multipass/6130/bin/check_kvm_support
Dec 08 09:09:56 cody-MS-7850 multipassd[26866]: process working dir '/snap/multipass/6130/qemu'
Dec 08 09:09:56 cody-MS-7850 multipassd[26866]: process program 'qemu-system-x86_64'
Dec 08 09:09:56 cody-MS-7850 multipassd[26866]: process arguments '--enable-kvm, -cpu, host, -nic, tap,ifname=tap-71c971542d0,script=no,downscript=no,model=virtio-net-pci,mac=52:54:00:1b:1a:06, -device, virtio-scsi-pci,id=scsi0, -drive, file=/var/snap/multipass/common/data/multipassd/vault/instances/datashyft-customer-portal/ubuntu-18.04-server-cloudimg-amd64.img,if=none,format=qcow2,discard=unmap,id=hda, -device, scsi-hd,drive=hda,bus=scsi0.0, -smp, 1, -m, 1024M, -qmp, stdio, -chardev, null,id=char0, -serial, chardev:char0, -nographic, -cdrom, /var/snap/multipass/common/data/multipassd/vault/instances/datashyft-customer-portal/cloud-init-config.iso, -loadvm, suspend, -machine, pc-i440fx-focal'
Dec 08 09:09:56 cody-MS-7850 multipassd[26866]: Resuming from a suspended state
Dec 08 09:09:56 cody-MS-7850 multipassd[26866]: Applied AppArmor policy: multipass.datashyft-customer-portal.qemu-system-x86_64
Dec 08 09:09:56 cody-MS-7850 multipassd[26866]: process state changed to Starting
Dec 08 09:09:56 cody-MS-7850 multipassd[26866]: process state changed to Running
Dec 08 09:09:56 cody-MS-7850 multipassd[26866]: [27407] started: qemu-system-x86_64 --enable-kvm -cpu host -nic tap,ifname=tap-71c971542d0,script=no,downscript=no,model=virtio-net-pci,mac=52:54:00:1b:1a:06 -device virtio-scsi-pci,id=scsi0 -drive file=/var/snap/multipass/common/data/multipassd/vault/instances/datashyft-customer-portal/ubuntu-18.04-server-cloudimg-amd64.img,if=none,format=qcow2,discard=unmap,id=hda -device scsi-hd,drive=hda,bus=scsi0.0 -smp 1 -m 1024M -qmp stdio -chardev null,id=char0 -serial chardev:char0 -nographic -cdrom /var/snap/multipass/common/data/multipassd/vault/instances/datashyft-customer-portal/cloud-init-config.iso -loadvm suspend -machine pc-i440fx-focal
Dec 08 09:09:56 cody-MS-7850 multipassd[26866]: process started
Dec 08 09:09:56 cody-MS-7850 multipassd[26866]: Waiting for SSH to be up
Dec 08 09:09:56 cody-MS-7850 multipassd[26866]: QMP: {"QMP": {"version": {"qemu": {"micro": 1, "minor": 2, "major": 4}, "package": "Debian 1:4.2-3ubuntu6.18"}, "capabilities": ["oob"]}}
Dec 08 09:09:57 cody-MS-7850 multipassd[26866]: qemu-system-x86_64: warning: TSC frequency mismatch between VM (3300034 kHz) and host (3299996 kHz), and TSC scaling unavailable
Dec 08 09:09:57 cody-MS-7850 multipassd[26866]:
Dec 08 09:09:57 cody-MS-7850 multipassd[26866]: QMP: {"return": {}}
Dec 08 09:14:29 cody-MS-7850 multipassd[26866]: QMP: {"timestamp": {"seconds": 1638983669, "microseconds": 971702}, "event": "RTC_CHANGE", "data": {"offset": -6149}}
```

The instance I've been trying to start is now in State: Unknown

Any assistance is greatly appreciated - please let me know if you need more info

**Comment by Saviq** [comment]
Hey @csandwith, sorry that you encountered this. Nothing looks immediately wrong from your logs.

Can you launch new instances?

You could try switching to the libvirt driver (`sudo apt install libvirt-daemon-system; sudo snap connect multipass:libvirt; sudo multipass set local.driver=libvirt`) and use `virt-manager` to see the instances' console output.

**Comment by townsend2010** [comment]
Hi @csandwith,

Yeah, it may indeed be related to #2316.  One way to see is to delete the snapshot in the instance image.

Disable the Multipass snap to ensure nothing is using the instance images:
`$ snap disable multipass`

Go to the directory of an instance image.  For example:
`$ cd /var/snap/multipass/common/data/multipassd/vault/instances/datashyft-customer-portal`

Check to make sure there is a snapshot saved in the image:
`$ qemu-img snapshot -l ubuntu-18.04-server-cloudimg-amd64.img`
You should see one named `suspend`.

Delete this snapshot:
`$ sudo qemu-img snapshot -d suspend ubuntu-18.04-server-cloudimg-amd64.img`

At this point you can either check that this instance will boot by re-enabling the Multipass snap or complete the `qemu-img snapshot` steps for any other non-booting instances.

To re-enable the Multipass Snap:
`$ snap enable multipass`

**Comment by csandwith** [comment]
@Saviq 

Well, interesting. I ran through the steps you gave (which required a bit more effort than just running them) and, despite the multipass snap taking a considerable amount of time to start afterwards, I was able to start my instance. I didn't have to use virt-manager (which isn't installed on my system anyway). Seems that switching over to libvirt did the trick though. Thank you.

@townsend2010 Thank you for the advice, but I think my issue is resolved
